hydradx/tests/test_omnipool_amm.py

```diff
@@ -1375,6 +1317,70 @@ def test_dynamic_fees(hdx_price: float):
         raise AssertionError('LRNA fee should decrease with time.')
 
 
+def test_dynamic_fee_multiple_block_update():
+    init_vol_out = mpf(100)
+    init_vol_in = 0
+    W = mpf(0.2)
+    amplification = 1
+    decay = 1 / mpf(100000)
+    init_liq = mpf(10000)
+    init_liq_oracle = mpf(10000)
+    R = init_liq - init_vol_out
+    num_blocks = 1000
+    init_fee = mpf(0.0025)
+    fee_min = mpf(0.0025)
+    fee_max = mpf(0.1)
+    fee = init_fee
+    vol_out_oracle = init_vol_out
+    vol_in_oracle = init_vol_in
+    liquidity_oracle = W * (init_liq - init_vol_out) + (1-W) * init_liq_oracle
+    for j in range(num_blocks):
+        x = (vol_out_oracle - vol_in_oracle) / liquidity_oracle
+        delta_fee = amplification * x - decay
+        fee = min(max(fee + delta_fee, fee_min), fee_max)
+        # oracle updates
+        vol_out_oracle = vol_out_oracle * (1 - W)
+        vol_in_oracle = vol_in_oracle * (1 - W)
+        liquidity_oracle = W * R + (1 - W) * liquidity_oracle
+
+    liquidity_oracle = W * (init_liq - init_vol_out) + (1-W) * init_liq_oracle
+    decays = [mpmath.power(1-W, i) for i in range(num_blocks)]
+    mult = sum([decays[j] / (1 + (liquidity_oracle - R)/R * decays[j]) for j in range(num_blocks)])
+    delta_fee = amplification * (init_vol_out - init_vol_in) / R * mult - decay * num_blocks
+    fee2 = min(max(init_fee + delta_fee, fee_min), fee_max)
+    assert fee == pytest.approx(fee2, rel=1e-20)
```

Comment:
Is this first test just the code of my original test? I don't think we need it anymore, at least not in integration tests.